src/control/cmd/daos_agent/fabric.go

```diff
@@ -141,24 +141,43 @@ func (n *NUMAFabric) GetDevice(numaNode int, netDevClass uint32, provider string
 	n.mutex.Lock()
 	defer n.mutex.Unlock()
 
-	if n.getNumNUMANodes() == 0 {
-		n.log.Infof("No fabric interfaces found, using default interface %q", DefaultFabricInterface.Name)
-		return DefaultFabricInterface, nil
-	}
-
 	fi, err := n.getDeviceFromNUMA(numaNode, netDevClass, provider)
 	if err == nil {
-		return fi, nil
+		return copyFI(fi, provider), nil
 	}
 
-	fi, err = n.findOnRemoteNUMA(netDevClass, provider)
+	fi, err = n.findOnAnyNUMA(netDevClass, provider)
 	if err != nil {
 		return nil, err
 	}
 
+	return copyFI(fi, provider), nil
+}
+
+func copyFI(fi *FabricInterface, provider string) *FabricInterface {
 	fiCopy := new(FabricInterface)
 	*fiCopy = *fi
-	return fiCopy, nil
+	if !providerIsVerbs(provider) {
+		// for non-verbs providers, including the domain can cause issues
+		fiCopy.Domain = ""
```

Comment:
What sort of issues? I'm a little wary of this... Seems like something that could come back to haunt us. Wouldn't it be better to address whatever the problems are?

Reply:
In the previous code we weren't setting the domain for non-verbs providers either. This change just centralizes the check.

If we always pass through the domain, CART would fail to initialize on non-verbs providers (e.g. ofi+sockets) saying the provider wasn't supported on the domain. If you don't pass the domain, CART sets it under the covers to the same as the interface name, and Mercury is happy with that.

Reply:
Hmm. Still not following exactly. I understand that this restores the previous behavior, but I'm questioning the correctness of the previous behavior. In the case of a non-verbs provider, what is the domain set to before it's cleared here?

The main thing I'm concerned about is hinging so much logic on verbs vs non-verbs, as it feels brittle (I do understand that there's lot of other brittleness in here, too). Just wondering if there's more we can do here to future-proof, or if we should just defer on those sorts of improvements until we refactor things.

Reply:
Before it's cleared, it's whatever "alias" value came out of hwloc. For interfaces that don't support verbs at all, like an ordinary ethernet port, typically nothing is returned. For those that support verbs, like Infiniband cards, we see something like "mlx5_0" or "hfi1_0". The issue shows up on interfaces that can use verbs, when we opt to use ofi+sockets on them instead. It's reliably reproducible if we don't exclude the domain at some level in that case.

I don't really know enough about the network stuff yet to say why the expectations are as they are. I have a suspicion that maybe we need to rethink how netdetect returns the fabric scan results in general. Right now it doesn't even include the domain, and we need to fetch it separately and make decisions based on that. netdetect really should be the one providing us with an appropriate domain in the fabric scan, based on the provider. That's a whole can of worms, though. I'd rather put it off for a separate PR.